pkg/reconciler/openshift/common/cabundle.go

```diff
@@ -0,0 +1,167 @@
+				break
+			}
+		}
+	}
+
+	// Let's add the trusted and service CA bundle ConfigMaps as a volume in
+	// the PodSpec which will later be mounted to add certs in the pod.
+	volumes = append(volumes,
+		corev1.Volume{
+			Name: trustedCAConfigMapVolume,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{Name: trustedCAConfigMapName},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  trustedCAKey,
+							Path: trustedCAKey,
+						},
+					},
+				},
+			},
+		},
+		corev1.Volume{
+			Name: serviceCAConfigMapVolume,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{Name: serviceCAConfigMapName},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  serviceCAKey,
+							Path: serviceCAKey,
+						},
+					},
+				},
+			},
+		})
+	deployment.Spec.Template.Spec.Volumes = volumes
+
+	// Now that the injected certificates have been added as a volume, let's
+	// mount them via volumeMounts in the containers
+	for i, c := range deployment.Spec.Template.Spec.Containers {
+		volumeMounts := c.VolumeMounts
+
+		// If volume mounts for injected certificates already exist then remove it
+		for _, volumeName := range []string{trustedCAConfigMapVolume, serviceCAConfigMapVolume} {
+			for i, vm := range volumeMounts {
+				if vm.Name == volumeName {
+					volumeMounts = append(volumeMounts[:i], volumeMounts[i+1:]...)
+					break
+				}
+			}
+		}
+
+		// /etc/ssl/certs is the default place where CA certs reside in *nix
+		// however this can be overridden using SSL_CERT_DIR, let's check for
+		// that here.
+		sslCertDir := "/etc/ssl/certs"
+		for _, env := range c.Env {
+			if env.Name == "SSL_CERT_DIR" {
+				sslCertDir = env.Value
+			}
+		}
+
+		// Let's mount the certificates now.
+		volumeMounts = append(volumeMounts,
+			corev1.VolumeMount{
+				Name:      trustedCAConfigMapVolume,
+				MountPath: filepath.Join(sslCertDir, trustedCAKey),
+				SubPath:   trustedCAKey,
+				ReadOnly:  true,
+			},
+			corev1.VolumeMount{
+				Name:      serviceCAConfigMapVolume,
+				MountPath: filepath.Join(sslCertDir, serviceCAKey),
+				SubPath:   serviceCAKey,
+				ReadOnly:  true,
+			},
+		)
+		c.VolumeMounts = volumeMounts
+		deployment.Spec.Template.Spec.Containers[i] = c
+	}
+
+	deployment.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   appsv1.SchemeGroupVersion.Group,
+		Version: appsv1.SchemeGroupVersion.Version,
+		Kind:    "Deployment",
+	})
+	m, err := toUnstructured(deployment)
+	if err != nil {
+		return err
+	}
+	u.SetUnstructuredContent(m.Object)
+	return nil
+}
+
+func toUnstructured(v interface{}) (*unstructured.Unstructured, error) {
+	b, err := json.Marshal(v)
+	if err != nil {
+		return nil, err
+	}
+	ud := &unstructured.Unstructured{}
+	if err := json.Unmarshal(b, ud); err != nil {
+		return nil, err
+	}
+	return ud, nil
+}
```

Comment:
nit: I think at some point, we will need to refactor these to work with full types in most case, hiding the unstructured mess from the important impl. detail.